sn_evm/src/data_payments.rs

```diff
@@ -244,17 +244,29 @@ impl PaymentQuote {
             return false;
         }
 
+        // TODO: Double check if this applies, as this will prevent a node restart with same ID
+        if new_quote.quoting_metrics.received_payment_count
+            < old_quote.quoting_metrics.received_payment_count
+        {
+            info!("claimed received_payment_count out of sequence");
+            return false;
+        }
+
         let old_elapsed = if let Ok(elapsed) = old_quote.timestamp.elapsed() {
             elapsed
         } else {
-            info!("timestamp failure");
-            return false;
+            // The elapsed call could fail due to system clock change
+            // hence consider the verification succeeded.
+            info!("old_quote timestamp elapsed call failure");
+            return true;
         };
         let new_elapsed = if let Ok(elapsed) = new_quote.timestamp.elapsed() {
             elapsed
         } else {
-            info!("timestamp failure");
-            return false;
+            // The elapsed call could fail due to system clock change
+            // hence consider the verification succeeded.
+            info!("new_quote timestamp elapsed call failure");
+            return true;
```

Comment:
Not 100% sure about this, @mickvandijke could you confirm this makes sense?

Reply:
I don't really understand why these checks are done at all. Maybe I'm missing something. But wouldn't we only need to verify a quote's signature to check if it is made by this peer? Why do we need to store historical quotes?

Reply:
This was to confirm a quote given by other node are not faked.

Reply:
The quote signature being valid already rules out that the quote could have been faked by another node or client. So I think we can get rid of this.

This might be out of scope for this PR though, we can do it in a follow-up PR.

Reply:
> This was to confirm a quote given by other node are not faked.

Sry, I don't mean `faked by other`, I mean `faked by the quoter itself` .
i.e. a node gives a quote with out_of_sequence data to gaim higher chance to be picked as a payee.